packages/ui/src/elements/Thumbnail/index.tsx

```diff
@@ -77,33 +79,35 @@ export function ThumbnailComponent(props: ThumbnailComponentProps) {
 
   const classNames = [baseClass, `${baseClass}--size-${size || 'medium'}`, className].join(' ')
 
-  React.useEffect(() => {
+  const src = React.useMemo(() => {
     if (!fileSrc) {
+      return null
+    }
+
+    /**
+     * If an imageCacheTag is provided, append it to the fileSrc
+     * Check if the fileSrc already has a query string, if it does, append the imageCacheTag with an ampersand
+     */
+    const queryChar = fileSrc?.includes('?') ? '&' : '?'
+    return imageCacheTag ? `${fileSrc}${queryChar}${encodeURIComponent(imageCacheTag)}` : fileSrc
```

Comment:
I believe this will encode false for no reason here? We should first check for the cache tag.